### ISSUE TITLE
Possibility to use 2 solenoids

### ISSUE THREAD

**Issue by alextrical** [robin7331/pixel-pump-mainboard#3]
After looking over the design, I was slightly confused by the pneumatic plumbing of the device, hopefully I could make a suggestion to simplify the setup of the air path.

You should be able to use a single [4V110-06](https://www.aliexpress.com/item/4001216228059.html) intended for a pneumatic pistons, but use it in reverse, though its rather large, and will consume 3-5W of power 
Alternatively 2 exhausted/3-way solenoids to flip the air flow, similar to [these ](https://www.aliexpress.com/item/1005005497520090.html)

I can sketch up a pneumatic circuit if you would like, to clarify how I believe it should be possible to plumb to achieve a reversible air flow

**Comment by robin7331** [comment]
Appreciate your input. There were more criterie involved into the selection of valves than just simplification. It hat to fit inside the Pixel Pump, be compatible with the 2mm/4mm silicone hoses that were used everywhere else, etc. 

So the 4V110-06 type of valve assembly does not meet all those requirements. 

> Alternatively 2 exhausted/3-way solenoids to flip the air flow, similar to [these](https://www.aliexpress.com/item/1005005497520090.html)

Could you do a simple hand drawing how you intend to connect those?

**Comment by alextrical** [comment]
Agreed that the 4V110-06 is too big for a project like this, though it's more of an example of a device to look out for (if they exist on a smaller scale)

![20240520_064449.jpg](https://github.com/robin7331/pixel-pump-mainboard/assets/35117191/93dae0c7-bc65-4f9f-93b3-2274d6a1a7bd)

I know the icons aren't exactly correct for pneumatics, but hopefully it's understandable. I've used a switch icon in place of the vented valves.
I believe that connecting a valve on either side of the pump, and a tee joining both values should do the pneumatics.
When the pump is running. Energising either solenoid will apply either a vacuum or pressure.

Under normal conditions I would expect that when one solenoid is activated, the other one will not be.

**Comment by robin7331** [comment]
> though it's more of an example of a device to look out for (if they exist on a smaller scale)

I spent quite some time on sourcing and finding the right components for the pneumatics. I haven't found a better solution for the Pixel Pump than what I currently have. I'm open for suggestions if you find any other solution thats better (e.g. cheaper while still being good quality, smaller, simpler, etc.) 😋


> I know the icons aren't exactly correct for pneumatics, but hopefully it's understandable. I've used a switch icon in place of the vented valves.

I do see problems. Assuming the third, unused port of the valves is shut tight (which it has to be according to your drawing or you'll lose air pressure on the handle if it vents out the other valve) then there is no way in implementing the "pressure relief" functionality which is very important IMHO. Not only for the vacuum side but also for the positive pressure side.

**Comment by alextrical** [comment]
> I'm open for suggestions if you find any other solution thats better (e.g. cheaper while still being good quality, smaller, simpler, etc.) 😋

Hopefully I can find something that can help this awesome project

> I do see problems. Assuming the third, unused port of the valves is shut tight... then there is no way in implementing the "pressure relief" functionality

That's correct, there is no direct path for the pen's airline to atmospheric pressure, without going via the pump. (other than the tip of the pen itself)  A 3rd identical solenoid could be used if need be, it would at least simplify the BOM by reducing the amount of different parts to source

I have yet to build a Vacuum Pick Pen, so I'm currently working on the following assumptions:
On button hold, A sustained vacuum is needed to lift and hold a part
On release, a positive pressure is required to blow the part off the tip (for a few milliseconds)

Wouldn't the pressure relief be able to be achieved by running the valves in positive pressure configuration. 

I suppose the question I should be asking is, what benefit does the "pressure relief" functionality add to the product?
